extension/src/experiments/index.ts

```diff
@@ -759,4 +761,56 @@ export class Experiments extends BaseRepository<TableData> {
     const columnLikes = addStarredToColumns(columns)
     return pickColumnToFilter(columnLikes)
   }
+
+  private async patchStudioApiTimingIssue(ids: string[]) {
+    if (!this.studio.isConnected()) {
+      return
+    }
+
+    const [, { lsRemoteOutput }] = await Promise.all([
+      this.waitForStudioUpdate(),
+      this.waitForRemoteUpdate()
+    ])
+
+    const remoteExpShas = collectRemoteExpShas(lsRemoteOutput)
+
+    const shas = []
+    for (const { id, sha } of this.experiments.getExperiments()) {
+      if (ids.includes(id) && sha && remoteExpShas.has(sha)) {
+        shas.push(sha)
+      }
+    }
+
+    this.experiments.assumePushed(shas)
+  }
+
+  private waitForStudioUpdate() {
```

Comment:
[F] I tried to combine the `waitForStudioUpdate` & `waitForRemoteUpdate` methods but the code was a lot harder to read.